heron/scheduler-core/src/java/com/twitter/heron/scheduler/ScalableScheduler.java

```diff
@@ -0,0 +1,19 @@
+// Copyright 2016 Twitter. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+package com.twitter.heron.scheduler;
+
+public interface ScalableScheduler {
+  void addContainers(Integer count);
+  void removeContainers(Integer existingContainerCount, Integer count);
```

Comment:
That's the current impl, but we might need to rethink this. For aurora if we remove a container that's not at the end, I'm not sure what the container ids would do and there are places where a continuous set of ids is expected so it feels safer to remove from the end. But it is presumptuous to say the last ones should go.

When doing the packing diff we could pull out the container ids that are now empty and pass those to removeContainers? We'd need to make sure there isn't a mess if we remove containers not at the end though.